pygmt/src/hlines.py

```diff
@@ -0,0 +1,146 @@
+"""
+hlines - Plot horizontal lines.
+"""
+import numpy as np
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    build_arg_string,
+    data_kind,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    N="no_clip",
+    V="verbose",
+    W="pen",
+    l="label",
+    p="perspective",
+    t="transparency",
+)
+@kwargs_to_strings(p="sequence")
+def hlines(self, y=None, xmin=None, xmax=None, **kwargs):
+    """
+    Plot one or a collection of horizontal lines.
+    Takes a single y value or a list of individual y values and optionally
+    lower and upper x value limits as input.
+    Must provide *y*.
+    If y values are given without x limits then the current map boundaries are
+    used as lower and upper limits. If only a single set of x limits is given
+    then all lines will have the same length, otherwise give x limits for each
+    individual line. If only a single label is given then all lines are grouped
+    under this label in the legend (if shown). If each line should appear as a
+    single entry in the legend, give corresponding labels for all lines
+    (same for **pen**).
+    Parameters
+    ----------
+    y : float or 1d array
+        The y coordinates or an array of y coordinates of the
+        horizontal lines to plot.
+    no_clip : bool or str
+        ``'[c|r]'``.
+        Do NOT clip lines that fall outside map border [Default plots
+        lines whose coordinates are strictly inside the map border only].
+        The option does not apply to lines which are always
+        clipped to the map region. For periodic (360-longitude) maps we
+        must plot all lines twice in case they are clipped by the
+        repeating boundary. ``no_clip=True`` will turn off clipping and not
+        plot repeating lines. Use ``no_clip="r"`` to turn off clipping
+        but retain the plotting of such repeating lines, or use
+        ``no_clip="c"`` to retain clipping but turn off plotting of
+        repeating lines.
+    {W}
+    {V}
+    label : str
+        Add a legend entry for the line being plotted.
+    {p}
+    {t}
+        *transparency* can also be a 1d array to set varying transparency
+        for lines.
+
+    """
+
+    kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
+
+    y = np.atleast_1d(y)
+    list_length = len(y)
+
+    # prepare x values
+    def prep_data(xmin, xmax, list_length):
+
+        if xmin is None and xmax is None:
+            with Session() as lib:
+                # get limits from current map boundings if not given
+                # via xmin, xmax
+                x = np.array([[lib.extract_region()[0]], [lib.extract_region()[1]]])
+                x = np.repeat(x, list_length, axis=1)
```

Comment:
Use the `Figure.region` property (https://www.pygmt.org/dev/api/generated/pygmt.Figure.html#pygmt-figure):
```suggestion
            # get limits from current map boundings if not given via xmin, xmax
            xmin, xmax = fig.region[0:2]
            x = np.repeat([[xmin], [xmax]], list_length, axis=1)
```